### PR TITLE
Add Async generators

### PR DESCRIPTION
**Pull request by Elchi3**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function*

### CODE REVIEW

File: features/async-generators.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Async generators
+description: Async generator functions (`async function*`) are asynchronous and always yield `Promise` objects.
+group: javascript
```
Comment (jamesnw) [inline features/async-generators.yml:3]: Would it make sense to have a `generators` group?

File: features/async-generators.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Async generators
+description: Async generator functions (`async function*`) are asynchronous and always yield `Promise` objects.
```
Comment (jamesnw) [inline features/async-generators.yml:2]: I think "are asynchronous" is redundant to "Async". Do we want to explain generators here as well? Perhaps 

```suggestion
description: Async generator functions (`async function*`) create iterators that return multiple Promises, one after another, on-demand.
```

**Review by jamesnw — APPROVED** [review]
Pre-review. Only non-blocking suggestions.

File: features/async-generators.yml (hunk 1 above)
Comment (Elchi3) [reply features/async-generators.yml:3]: Let me add that as a follow up when this PR and https://github.com/web-platform-dx/web-features/pull/1664 have been merged

File: features/async-generators.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Async generators
+description: Async generator functions (`async function*`) create iterators that return multiple Promises, one after another, on-demand.
```
Comment (ddbeck) [inline features/async-generators.yml:2]:
```suggestion
description: Async generator functions (`async function*`) create iterators that return multiple promises, one after another, on-demand.
```

**Review by ddbeck — APPROVED** [review]

File: features/async-generators.yml (hunk 1 above)
Comment (ddbeck) [reply features/async-generators.yml:3]: These are both merged now. 👍

**Pull request merged**